src/Algebra/Morphism/Construct/Initial.agda

```diff
@@ -0,0 +1,73 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- The unique morphism from the initial object,
+-- for each of the relevant categories. Since
+-- `Semigroup` and `Band` are simply `Magma`s
+-- satisfying additional properties, it suffices to
+-- define the morphism on the underlying `RawMagma`.
+------------------------------------------------------------------------
+
+{-# OPTIONS --cubical-compatible --safe #-}
+
+open import Level using (Level)
+
+module Algebra.Morphism.Construct.Initial {c ℓ : Level} where
+
+open import Algebra.Bundles using (Magma)
+open import Algebra.Bundles.Raw using (RawMagma)
+open import Algebra.Construct.Initial {c} {ℓ}
+import Algebra.Morphism.Definitions as MorphismDefinitions
+open import Algebra.Morphism.Structures using (module MagmaMorphisms)
+open import Function.Definitions using (Injective)
+import Relation.Binary.Morphism.Definitions as Definitions
+open import Relation.Binary.Morphism.Structures
+
+private
+  variable
+    m ℓm : Level
+
+
+------------------------------------------------------------------------
+-- The underlying data of the morphism
+
+module UniqueMorphism (M : RawMagma m ℓm) where
+
+  open RawMagma M
+  open MorphismDefinitions ℤero.Carrier Carrier _≈_
+
+  zero : ℤero.Carrier → Carrier
+  zero ()
+
+  cong : Definitions.Homomorphic₂ ℤero.Carrier Carrier ℤero._≈_ _≈_ zero
+  cong {x = ()}
+
+  isRelHomomorphism : IsRelHomomorphism ℤero._≈_ _≈_ zero
+  isRelHomomorphism = record { cong = cong }
+
+  homo : Homomorphic₂ zero ℤero._∙_ _∙_
+  homo ()
+
+  injective : Injective ℤero._≈_ _≈_ zero
+  injective {x = ()}
+
+------------------------------------------------------------------------
+-- Magma
+
+module _ (M : Magma m ℓm) where
```

Comment:
Can't/shouldn't this and all the similar modules in `Terminal` be take raw versions of the bundle as parameters instead?

You could then save a lot of lines (and IMO maybe make it easier to read?) by opening `MagmaMorphisms` etc. at the top of the file without a parameter?

Reply:
Hmmm... that's an interesting/tricky one. Clearly, for each of `Initial` and `Terminal`, there is the underlying module parameterised on a `Raw` bundle.

For whatever reason (?), I associate `Construct`s as being parametrised on the *bundle*, proxying for the 'name of the category'. But you're quite correct that all the other modules in that directory only parametrise on the `Raw` thing (?), so for consistency's sake, I should revise as you suggest.

Reply:
Not sure I understand how to save the lines though. Will push the re-parametrisation, and then maybe you could offer me a steer on that?

Reply:
I've rearranged it to get rid of almost 70 lines of boilerplate. Let me know if you think I've lost anything crucial?

Reply:
Looks good, modulo it now being harder for me to read, but that's a common problem! 

Your desire for brevity contrasts with what I regard as mine for (greater!?) clarity, but it does seem to simply be a contrast in styles. 

Happy to let yours carry the day. Not sure how well I can internalise it though! ;-)